File: src/main/java/io/tatum/model/response/offchain/Address.java
package io.tatum.model.response.offchain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.math.BigInteger;

/**
 * The type Address.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode
@JsonIgnoreProperties(ignoreUnknown = true)
public class Address {
    /**
     * Blockchain address.
     *
     * @type {string}
     * @memberof Address
     */
    private String address;

    /**
     * Currency of generated address. BTC, LTC, BCH, ETH, XRP, ERC20.
     *
     * @type {string}
     * @memberof Address
     */
    private String currency;

    /**
     * Derivation key index for given address.
     *
     * @type {number}
     * @memberof Address
     */
    private int derivationKey;

    /**
     * Extended public key to derive address from. In case of XRP, this is account address,
     * since address is defined as DestinationTag, which is address field. In case of XLM, this is account address, since address is defined as message, which is address field.
     *
     * @type {string}
     * @memberof Address
     */
    private String xpub;

    /**
     * In case of XRP, destinationTag is the distinguisher of the account.
     *
     * @type {number}
     * @memberof Address
     */
    private BigInteger destinatinTag;

    /**
     * In case of XLM, message is the distinguisher of the account.
     *
     * @type {string}
     * @memberof Address
     */
    private String message;
}

File: src/main/kotlin/model/response/offchain/Address.kt
package model.response.offchain

import java.math.BigInteger

/**
 * @property address Blockchain address.
 * @property currency Currency of generated address. BTC, LTC, BCH, ETH, XRP, ERC20.
 * @property derivationKey Derivation key index for given address.
 * @property xpub Extended public key to derive address from. In case of XRP, this is account address,
 * @property destinatinTag In case of XRP, destinationTag is the distinguisher of the account.
 * @property message In case of XLM, message is the distinguisher of the account.
 */
class Address(
    val address: String,
    val currency: String,
    val derivationKey: BigInteger? = null,
    val xpub: String? = null,
    val destinatinTag: BigInteger? = null,
    val message: String? = null,
)
